gateway-ha/src/main/java/io/trino/gateway/ha/resource/EntityEditorResource.java

```diff
@@ -29,150 +29,156 @@
-          break;
-        case SELECTOR:
-          SelectorsDetail selectorDetails = OBJECT_MAPPER.readValue(jsonPayload,
-              SelectorsDetail.class);
-          List<SelectorsDetail> oldSelectorDetails =
-              resourceGroupsManager.readSelector(selectorDetails.getResourceGroupId(), database);
-          if (oldSelectorDetails.size() >= 1) {
-            resourceGroupsManager.updateSelector(oldSelectorDetails.get(0),
-                selectorDetails, database);
-          } else {
-            resourceGroupsManager.createSelector(selectorDetails, database);
-          }
-          break;
-        default:
-      }
-    } catch (IOException e) {
-      log.error(e.getMessage(), e);
-      throw new WebApplicationException(e);
-    }
-    return Response.ok().build();
-  }
-
-  @GET
-  @Path("/{entityType}")
-  @Produces(MediaType.APPLICATION_JSON)
-  public Response getAllEntitiesForType(@PathParam("entityType") String entityTypeStr,
-                                        @QueryParam("useSchema") String database) {
-    EntityType entityType = EntityType.valueOf(entityTypeStr);
-
-    switch (entityType) {
-      case GATEWAY_BACKEND:
-        return Response.ok(gatewayBackendManager.getAllBackends()).build();
-      case RESOURCE_GROUP:
-        return Response.ok(resourceGroupsManager.readAllResourceGroups(database)).build();
-      case SELECTOR:
-        return Response.ok(resourceGroupsManager.readAllSelectors(database)).build();
-      default:
+    public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger log = LoggerFactory.getLogger(EntityEditorResource.class);
+    @Inject
+    private GatewayBackendManager gatewayBackendManager;
+    @Inject
+    private ResourceGroupsManager resourceGroupsManager;
+
+    @Inject
+    private RoutingManager routingManager;
+
+    @GET
+    @Produces(MediaType.TEXT_HTML)
+    public EntityView entityUi(@Context SecurityContext securityContext)
+    {
+        return new EntityView("/template/entity-view.ftl", securityContext);
     }
-    return Response.ok(ImmutableList.of()).build();
-  }
-
-  public static class EntityView extends View {
-    private String displayName;
 
-    protected EntityView(String templateName, SecurityContext securityContext) {
-      super(templateName, Charset.defaultCharset());
-      setDisplayName(securityContext.getUserPrincipal().getName());
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public List<EntityType> getAllEntityTypes()
+    {
+        return Arrays.asList(EntityType.values());
     }
 
-    public String getDisplayName()
-    {return this.displayName;}
-
-    public void setDisplayName(String displayName)
-    {this.displayName = displayName;}
-
-    public boolean equals(final Object o)
+    @POST
+    public Response updateEntity(@QueryParam("entityType") String entityTypeStr,
```

Comment:
nit: Move the 1st argument to the next line.
```java
    @POST
    public Response updateEntity(
            @QueryParam("entityType") String entityTypeStr,
            @QueryParam("useSchema") String database,
            String jsonPayload)
```
Same for others.

Reply:
Done